[Feat/323] 모임 기록 구현

edv-Shin opened the pull request:
## Related Issue
- #323 

## Type of change
<!-- 작업의 종류를 선택해주세요. -->
- [x] Feature : 새로운 기능 추가

## PR Desciption
> 변경 사항 설명
- 모임기록 CRUD 비즈니스 로직 추가
- 모임기록 편집모드 ui 추가

## Requirements for Reviewer
> 리뷰어가 특별히 봐주었으면 하는 부분이 있다면 작성해주세요
> ex) 메서드 XXX의 이름을 더 잘 짓고 싶은데 혹시 좋은 명칭이 있을까요?

- EditMoimDiaryUsecase에서 일기/활동 중에 변화가 있는 것만 호출하고 싶었는데.. 일단 일기 변화 여부만 반영했습니다
- 모임기록의 일기랑 개인기록 CRUD가 같은 api를 사용해서 추후에 Usecase로 분리할 예정입니다

- 모임 기록 조회 시에 활동 조회와 활동 정산 조회가 다른 api를 사용하는데, 
활동을 리스트로 관리하다 보니 하나의 객체로 관리하는게 편해서 UseCase에서 하나의 모델로 합쳤습니다

- 활동 정산/참가자 다이얼로그에서는 모두 데이터바인딩을 사용하면 각각 LiveData를 만들어야해서.. 
양이 많지도 않으니 뷰바인딩쪽으로 틀었습니다

다이얼로그로 호출하는 수정 api들의 result 관찰을 SharedFlow를 사용했습니다
- 다이얼로그에서 activityViewModel을 사용하다보니, 한번 result를 받고나서 다이얼로그를 다시 킬 때 observe가 실행됨
- SharedFlow를 사용하면 일회성으로 관찰을 알리기 때문에 사용

## PR Log

> PR 작업하면서 고민했던 내용, 해결한 내용, 고민 중인 내용 등

### 새롭게 배운 것

- 

### 고민 중인 사항

- 서버 에러의 토스트 메세지를 띄울 때 runCatching의 onFailure로 가버리는 문제.. 어떻게 해결해야할지 고민입니다

## 첨부 자료

-

edv-Shin committed:
✨ Feat: 편집모드 ui 구현, 모임 기록 패키지 이동

Related to: #323

edv-Shin committed:
✨ Feat: 활동 참가자, 정산 다이얼로그 구현

Related to: #323

edv-Shin committed:
✨ Feat: 일기장 추가/삭제, 활동 삭제 구현

Related to: #323

edv-Shin committed:
✨ Feat: 모임기록 CRUD 비즈니스 로직 구현

Related to: #323

edv-Shin committed:
✨ Feat: 편집모드에서 조회모드로 전환 로직 구현

Related to: #323

edv-Shin committed:
✨ Feat: 전체 정산 구현

Related to: #323

edv-Shin committed:
♻️ Refactor: 활동 정산 api 스펙 변경

Related to: #323

edv-Shin committed:
🐛 Fix: 모임 총 정산 계산 로직 수정, 리스트가 보이지 않는 버그 수정

Related to: #323

edv-Shin committed:
✨ Feat: 추가/수정 시 로딩 구현

Related to: #323

edv-Shin committed:
♻️ Refactor: 모임 기록 다이얼로그 string으로 변경

Related to: #323

edv-Shin committed:
✨ Feat: 활동 시간을 기록 시간에 제한

Related to: #323

edv-Shin committed:
Merge remote-tracking branch 'origin/dev' into feat/moim-diary-323

edv-Shin committed:
✨ Feat: 활동 기간 제한 TimePicker 추가

Related to: #323

edv-Shin committed:
✨ Feat: 조회모드로 전환 시 날짜 선택해제

Related to: #323

edv-Shin committed:
🎨 Style: 불필요한 주석 삭제

Related to: #323

edv-Shin committed:
✨ Feat: 일정과 기록 연결

Related to: #323

edv-Shin committed:
✨ Feat: 커뮤니티의 모임 일정에서 기록 연결

Related to: #323

edv-Shin committed:
🎨 Style: 모임 참여자 캘린더 이동

Related to: #323

nahy-512 commented on app/src/main/java/com/mongmong/namo/presentation/ui/community/diary/ActivityParticipantsDialog.kt:
지금 community 안에 바로 diary가 있는 걸로 보이는데, 제가 커뮤니티쪽 작업하면서 초반에 생각했던 구조는
바텀네비에서 community가 있고, 커뮤니티 안에 크게 모임/친구 기능이 있고, 모임 안에 모임 일정/모임 기록이 있으니까 diary를 moim 안으로 넣는 게 어떨까 싶었어요..! 혹시 이 부분에 대해서 어떻게 생각하실까요?

nahy-512 replied:
그래서 현재 schedule도 moim 안에 있는 구조이긴 합니다!
(calendar의 경우 moim/friend에서 공통적으로 사용함)

edv-Shin replied:
그렇게 바꿔도 괜찮을 것 같네요. 수정해서 올리겠습니다.

edv-Shin committed:
🐛 Fix: 모임 기록 버그 수정

- 모임 기록 생성/삭제 후 뷰 갱신이 되지 않는 버그 수정
- 전체 정산 텍스트 색상 버그 수정
- 모임 기록 생성 뷰에서 이미지 추가 버튼이 보이지 않는 버그 수정

Related to: #323

edv-Shin committed:
🛠️ Chore: 모임 기록 패키지 이동

Related to: #323

edv-Shin committed:
✨ Feat: 활동이 존재하고 정산이 0원일 때 텍스트 변경

Related to: #323

edv-Shin committed:
🐛 Fix: 활동 참석자 전체 인원이 뜨지 않는 버그 수정

Related to: #323
